### PR TITLE
Modernize administrator course pages

### PR DESCRIPTION
**Pull request by nwalters512**
I'm doing this ahead of some changes to the course request page. It looks quite big, but as much as possible, this is all just mechanical conversion of EJS to TS.

### CODE REVIEW

File: apps/prairielearn/src/lib/course-request.sql [hunk 1]
```diff
@@ -50,7 +50,10 @@ FROM
   LEFT JOIN select_course_request_jobs AS j ON j.id = r.id
 WHERE
   $show_all = 'true'
-  OR r.approved_status NOT IN ('approved', 'denied');
+  OR r.approved_status NOT IN ('approved', 'denied')
+ORDER BY
+  r.created_at DESC,
+  r.id DESC;
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/course-request.sql:56]: Change in behavior: we now have consistent ordering of course requests, with the most recent ones first.

File: apps/prairielearn/src/migrations/20240301223248_course_requests__not_null.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ALTER COLUMN short_name
+SET NOT NULL;
+
+ALTER TABLE course_requests
+ALTER COLUMN title
+SET NOT NULL;
+
+ALTER TABLE course_requests
+ALTER COLUMN user_id
+SET NOT NULL;
```
Comment (nwalters512) [inline apps/prairielearn/src/migrations/20240301223248_course_requests__not_null.sql:15]: I checked in all prod environments and these were all already non-null, so I figured I'd enforce it at the schema level. This makes some usage of these columns safer.

File: apps/prairielearn/src/pages/administratorAdmins/administratorAdmins.html.ts [hunk 3]
```diff
@@ -46,8 +46,6 @@ export function AdministratorAdmins({
                     id: 'administratorInsertButton',
                   }),
                 )}"
-                data-trigger="manual"
-                onclick="$(this).popover('show')"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/administratorAdmins/administratorAdmins.html.ts:50]: This is the default behavior; I don't know why this is here. Maybe it's just been copy/pasted around forever? I removed it from all admin pages and a few others too.

File: apps/prairielearn/src/pages/administratorCourseRequests/administratorCourseRequests.ts [hunk 4]
```diff
@@ -10,24 +9,29 @@ import {
...
+      }),
+    );
   }),
 );
 
 router.post(
   '/',
   asyncHandler(async (req, res) => {
-    if (!res.locals.is_administrator) throw error.make(403, 'Insufficient permissions');
-
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/administratorCourseRequests/administratorCourseRequests.ts:30]: AFAICT this permissions check is unnecessary; we already have middleware sitting in front of all admin routes that checks for this.

File: apps/prairielearn/src/pages/administratorCourses/administratorCourses.html.ts [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+      </div>
+      <div class="form-group">
+        <label for="courseAddInputTimezone">Timezone:</label>
+        <input
+          type="text"
+          class="form-control"
+          id="courseAddInputTimezone"
+          name="display_timezone"
+          value="${institutions[0]?.display_timezone}"
+        />
```
Comment (mylesw27) [inline apps/prairielearn/src/pages/administratorCourses/administratorCourses.html.ts:265]: Do we want to consider converting this to a dropdown?

**Review by mylesw27 — APPROVED** [review]
Leaving a small question that can be addressed by a future PR. Just something to consider while we are here. Everything else looks good.

File: apps/prairielearn/src/pages/administratorCourses/administratorCourses.html.ts (hunk 5 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/administratorCourses/administratorCourses.html.ts:265]: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones?wprov=sfla1

The list is big...

Comment (nwalters512) [reply apps/prairielearn/src/pages/administratorCourses/administratorCourses.html.ts:265]: @jonatanschroeder we have prior art for this:

https://github.com/PrairieLearn/PrairieLearn/blob/479268edd99e96dbc4f47659333f0774c72ccef7/apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.html.ts#L95

@mylesw27 there's just enough complexity associated with this (specifically updating the selected timezone based on the selected institution) that I'll hold off on that for now. This form is also essentially never used in practice, so it shouldn't be too much of a usability issue.

**Pull request merged**